services/proctoring-service/ai-service/main.py

```
@@ -0,0 +1,98 @@
+    occlusion = 1 if (mean_b < OCCLUSION_DARK or std_b < OCCLUSION_UNIFORM) else 0
+    return {"person_count": person_count, "phone": phone_flag, "mean_b": mean_b, 
+            "std_b": std_b, "pitch": pitch, "yaw": yaw, "motion": motion, "occlusion": occlusion}
+
+# --- KHỞI TẠO SERVER VÀ TẢI MODEL ---
+app = FastAPI(title="Proctoring AI Service", version="2.0.0")
+yolo_model = YOLO('yolov8n.pt')
+mp_face_mesh = mp.solutions.face_mesh.FaceMesh(static_image_mode=True, max_num_faces=2, refine_landmarks=True)
+clf = None
+try:
+    with open("violation_model.pkl", "rb") as f:
+        clf = pickle.load(f)
```

Ruff (0.14.0)

`pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)